backend/app/api/routes/dde.py

```diff
@@ -214,33 +216,121 @@ async def extract_from_file(*, session: SessionDep, current_user: CurrentUser, n
-            ChatCompletionMessage(role="user", content=f"Here is the next input:\n####\nINPUT:{prompt}")  
+            ChatCompletionMessage(role="user", content=f"Maintenant, faites la même extraction sur un nouveau document d'input:\n####\nINPUT:{prompt}")  
         ]
     chat_completion_request = ChatCompletionRequest(   
-            model='gpt-4o',
+            model='gpt-4o-2024-08-06',
             messages=messages,
             max_tokens=2000,
             temperature=0.1,
             logprobs=True,
             top_logprobs= 5,
             
         ).model_dump(exclude_unset=True)
+    
     chat_completion_response = await ArenaHandler(session, current_user, chat_completion_request).process_request()
     extracted_info=chat_completion_response.choices[0].message.content
     # TODO: Improve the prompt to ensure the output is always a valid JSON
     json_string = extracted_info[extracted_info.find('{'):extracted_info.rfind('}')+1]
-    return json.loads(json_string)
+    extracted_data = {k: v for k, v in json.loads(json_string).items() if k not in ('source', 'year')}   
+    logprob_data = extract_logprobs_from_response(chat_completion_response, extracted_data)
+    return {'extracted_info': json.loads(json_string), 'logprob_data': logprob_data}
+
+def validate_extracted_text(text: str):
+    if text == "":
+        raise HTTPException(status_code=500, detail="The extracted text from the document is empty. Please check if the document is corrupted.")
+    
+    
+# TODO: Optimize the entire process of extracting and handling log probabilities from OpenAI for the identified tokens.
+def is_equal_ignore_sign(a, b) -> bool:  
+    try:
+        a = float(a)
+        b = float(b)
+    except ValueError:
+        return False
+    return abs(a) == abs(b)      # necessary because logits are associated only with numerical tokens, so here values are considered in their absolute form, ignoring the sign.
+
+def combined_token_in_extracted_data(combined_token, extracted_data) -> bool:  
+    try:
+        combined_token = float(combined_token)
+    except ValueError:
+        return False
+    return any(is_equal_ignore_sign(combined_token, value)  
+                for value in extracted_data if isinstance(value, (int, float)))
+
+def find_key_by_value(combined_token, extracted_data) -> str | None: 
+    try:
+        combined_token = float(combined_token)
+    except ValueError:
+        return None
+    return next((k for k, v in extracted_data.items() 
+                    if isinstance(v, (int, float)) and is_equal_ignore_sign(combined_token, v)), None)    
+
+def extract_logprobs_from_response(response: Any, extracted_data: dict[str, Any]) -> dict[str, float]:
+    logprob_data = {}
+    tokens_info = response.choices[0].logprobs.content
+
+    def process_numeric_values(extracted_data: dict, path=''):
+        i = 0
+        while i < len(tokens_info):    
```

Comment:
Why not use a "for loops" ? "While loops" may never end and it is not always straightforward to check.